### DISCUSSION TITLE
Skript Tasks: allows for functions to be scheduled with a Task Scheduler.

### DISCUSSION THREAD

**Discussion by Rakise** [SkriptLang/Skript#6230] [Suggestions]
The proposed Task Scheduler would enable users to create and manage tasks for specific functions

Key Features:

1. Task Creation:

> Users can create tasks to execute specific functions after a specified delay or under certain conditions.
`create task "MyTask":`
 `   wait 1.5 seconds`
`    set {variable} to false`
    

2.  Task Detection:

> The plugin would allow users to check whether a specific task is currently active or not.

`if task "MyTask" is active:`
 `   # Perform actions if the task is currently running`
    
3. Task Cancellation:

> Users can cancel a running task.

`cancel task "MyTask"`

obs: if a user decides to cancel a task while it is in progress, it should indeed interrupt any ongoing processes, including the "wait" effect.

**Comment by erenkarakal** [comment]
[check this out](https://github.com/Mr-Darth/Skriptness/blob/master/scripts/tools/horologe.sk)
